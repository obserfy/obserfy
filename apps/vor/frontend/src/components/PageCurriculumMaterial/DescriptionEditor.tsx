import { Trans } from "@lingui/macro"
import { FC, useState } from "react"
import { Button, Card, Flex } from "theme-ui"
import usePatchMaterial from "../../hooks/api/curriculum/usePatchMaterial"
import { ReactComponent as CloseIcon } from "../../icons/close.svg"
import Icon from "../Icon/Icon"
import MarkdownEditor from "../MarkdownEditor/MarkdownEditor"
import { Typography } from "../Typography/Typography"

const useCachedValue = (key: string, initialValue: string) => {
  let cachedValue: string | null = null
  if (typeof window !== "undefined") {
    cachedValue = localStorage.getItem(key)
  }
  const [value, setValue] = useState(cachedValue ?? initialValue)

  const handleValueChange = (newValue: string) => {
    localStorage.setItem(key, newValue)
    setValue(newValue)
  }

  const clearCache = () => {
    localStorage.removeItem(key)
  }

  return { value, setValue: handleValueChange, clearCache }
}

const DescriptionEditor: FC<{
  initialValue?: string
  onDismiss: () => void
  onSave: () => void
  materialId: string
  subjectId: string
}> = ({ materialId, subjectId, onDismiss, onSave, initialValue = "" }) => {
  const { value, setValue, clearCache } = useCachedValue(
    materialId,
    initialValue
  )
  const patchMaterial = usePatchMaterial(materialId, subjectId)

  const handleSave = async () => {
    try {
      await patchMaterial.mutateAsync({ description: value })
      clearCache()
      onSave()
    } catch (e) {
      Sentry.captureException(e)
    }
  }

  const handleCancel = () => {
    clearCache()
    onDismiss()
  }

  return (
    <Card variant="responsive">
      <Flex pt={3} pb={2}>
        <Typography.Body px={3} sx={{ fontWeight: "bold" }}>
          <Trans>Description</Trans>
        </Typography.Body>

        <Button variant="outline" ml="auto" p={1} mr={2} onClick={handleCancel}>
          <Icon as={CloseIcon} fill="danger" />
        </Button>

        <Button
          mr={3}
          sx={{ fontWeight: "bold", fontSize: 0 }}
          onClick={handleSave}
        >
          <Trans>Save</Trans>
        </Button>
      </Flex>

      <MarkdownEditor value={value} onChange={setValue} />
    </Card>
  )
}

export default DescriptionEditor
