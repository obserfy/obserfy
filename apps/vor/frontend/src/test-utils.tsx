/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
