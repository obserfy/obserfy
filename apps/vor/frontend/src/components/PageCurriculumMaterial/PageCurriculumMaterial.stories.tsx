import { Meta, Story } from "@storybook/react"
import PageCurriculumMaterial, {
  PageCurriculumMaterialProps,
} from "./PageCurriculumMaterial"

export default {
  title: "Core/PageCurriculumMaterial",
  component: PageCurriculumMaterial,
  parameters: {
    componentSubtitle: "Just a simple PageCurriculumMaterial",
  },
} as Meta

const Template: Story<PageCurriculumMaterialProps> = (args) => (
  <PageCurriculumMaterial {...args} />
)

export const Default = Template.bind({})
Default.args = {}
