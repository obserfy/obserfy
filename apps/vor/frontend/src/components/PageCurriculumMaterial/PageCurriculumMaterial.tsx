import { t, Trans } from "@lingui/macro"
import { FC, useState } from "react"
import { Box, Button, Card, ThemeUIStyleObject } from "theme-ui"
import useDeleteMaterial from "../../hooks/api/curriculum/useDeleteMaterial"
import useGetMaterial from "../../hooks/api/curriculum/useGetMaterial"
import usePatchMaterial from "../../hooks/api/curriculum/usePatchMaterial"
import { useGetArea } from "../../hooks/api/useGetArea"
import { useGetSubject } from "../../hooks/api/useGetSubject"
import useVisibilityState from "../../hooks/useVisibilityState"
import { ReactComponent as DeleteIcon } from "../../icons/trash.svg"
import {
  ADMIN_CURRICULUM_URL,
  ADMIN_URL,
  CURRICULUM_AREA_URL,
  CURRICULUM_SUBJECT_URL,
} from "../../routes"
import DataBox from "../DataBox/DataBox"
import { Dialog } from "../Dialog/Dialog"
import DialogHeader from "../DialogHeader/DialogHeader"
import Icon from "../Icon/Icon"
import Input from "../Input/Input"
import { navigate } from "../Link/Link"
import MultilineDataBox from "../MultilineDataBox/MultilineDataBox"
import TopBar, { breadCrumb } from "../TopBar/TopBar"
import Typography from "../Typography/Typography"
import DescriptionEditor from "./DescriptionEditor"

export interface PageCurriculumMaterialProps {
  sx?: ThemeUIStyleObject
  areaId: string
  subjectId: string
  materialId: string
}
const PageCurriculumMaterial: FC<PageCurriculumMaterialProps> = ({
  subjectId,
  areaId,
  materialId,
}) => {
  const area = useGetArea(areaId)
  const subject = useGetSubject(subjectId)
  const material = useGetMaterial(materialId)

  const descriptionEditor = useVisibilityState()
  const deleteDialog = useVisibilityState()

  return (
    <Box mx="auto" sx={{ maxWidth: "maxWidth.lg", width: "100%" }}>
      <TopBar
        containerSx={{ display: ["flex", "flex", "none"] }}
        breadcrumbs={[
          breadCrumb("Admin", ADMIN_URL),
          breadCrumb("Curriculum", ADMIN_CURRICULUM_URL),
          breadCrumb(area.data?.name ?? "", CURRICULUM_AREA_URL(areaId)),
          breadCrumb(
            subject.data?.name ?? "",
            CURRICULUM_SUBJECT_URL(areaId, subjectId)
          ),
          breadCrumb(material.data?.name ?? ""),
        ]}
      />

      <Box pt={[0, 0, 3]}>
        <NameDataBox
          name={material.data?.name ?? "..."}
          materialId={materialId}
          subjectId={subjectId}
        />

        {descriptionEditor.visible ? (
          <DescriptionEditor
            initialValue={material.data?.description}
            onDismiss={descriptionEditor.hide}
            onSave={descriptionEditor.hide}
            materialId={materialId}
            subjectId={subjectId}
          />
        ) : (
          <Card variant="responsive" pb={2}>
            <MultilineDataBox
              key={material.data?.description}
              label="Description"
              value={material.data?.description ?? ""}
              placeholder={t`Not set`}
              onEditClick={descriptionEditor.show}
            />
          </Card>
        )}

        <Button
          variant="outline"
          sx={{ color: "danger" }}
          ml="auto"
          mt={3}
          mr={3}
          onClick={deleteDialog.show}
        >
          <Icon as={DeleteIcon} mr={2} fill="danger" />
          <Trans>Delete material</Trans>
        </Button>
      </Box>

      {deleteDialog.visible && (
        <DeleteMaterialDialog
          materialId={materialId}
          name={material.data?.name}
          onDismiss={deleteDialog.hide}
          subjectId={subjectId}
          onDeleted={() => {
            navigate(CURRICULUM_SUBJECT_URL(areaId, subjectId))
          }}
        />
      )}
    </Box>
  )
}

const DeleteMaterialDialog: FC<{
  materialId: string
  subjectId: string
  name?: string
  onDismiss: () => void
  onDeleted: () => void
}> = ({ onDeleted, materialId, onDismiss, subjectId, name = "" }) => {
  const deleteMaterial = useDeleteMaterial(materialId, subjectId)

  const handleDelete = () => {
    try {
      deleteMaterial.mutate()
      onDeleted()
    } catch (e) {
      Sentry.captureException(e)
    }
  }

  return (
    <Dialog>
      <DialogHeader
        title={t`Delete Material`}
        onCancel={onDismiss}
        onAccept={handleDelete}
        onAcceptText={t`Delete`}
      />
      <Box p={3} sx={{ backgroundColor: "background" }}>
        <Typography.Body>
          <Trans>
            This will permanently delete ${name} and data related to it. Are you
            sure?
          </Trans>
        </Typography.Body>
      </Box>
    </Dialog>
  )
}

const NameDataBox: FC<{
  name: string
  materialId: string
  subjectId: string
}> = ({ name, materialId, subjectId }) => {
  const editDialog = useVisibilityState()

  return (
    <Box px={[0, 3]}>
      <Card
        mb={3}
        sx={{
          width: "100%",
          boxSizing: "border-box",
          borderRadius: [0, "default"],
        }}
      >
        <DataBox
          label="Material Name"
          value={name}
          onEditClick={editDialog.show}
        />
      </Card>
      {editDialog.visible && (
        <EditMaterialNameDialog
          initialValue={name}
          onDismiss={editDialog.hide}
          onSave={editDialog.hide}
          materialId={materialId}
          subjectId={subjectId}
        />
      )}
    </Box>
  )
}

const EditMaterialNameDialog: FC<{
  initialValue: string
  onSave: () => void
  onDismiss: () => void
  materialId: string
  subjectId: string
}> = ({ onSave, onDismiss, initialValue, materialId, subjectId }) => {
  const patchMaterial = usePatchMaterial(materialId, subjectId)
  const [value, setValue] = useState(initialValue)

  const handleSave = async () => {
    try {
      await patchMaterial.mutateAsync({ name: value })
      onSave()
    } catch (e) {
      Sentry.captureException(e)
    }
  }

  return (
    <Dialog>
      <DialogHeader
        title={t`Edit Material Name`}
        onCancel={onDismiss}
        onAccept={handleSave}
      />

      <Box p={3} sx={{ backgroundColor: "background" }}>
        <Input
          label="Name"
          value={value}
          onChange={(e) => setValue(e.target.value)}
          sx={{ width: "100%" }}
        />
      </Box>
    </Dialog>
  )
}

export default PageCurriculumMaterial
