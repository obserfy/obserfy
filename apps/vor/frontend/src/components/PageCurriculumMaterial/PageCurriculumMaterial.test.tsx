import { render } from "../../test-utils"
import PageCurriculumMaterial from "./PageCurriculumMaterial"

describe("PageCurriculumMaterial", () => {
  it("should render correctly", () => {
    const { container } = render(
      <PageCurriculumMaterial areaId="" materialId="" subjectId="" />
    )
    expect(container).toMatchSnapshot()
  })
})
