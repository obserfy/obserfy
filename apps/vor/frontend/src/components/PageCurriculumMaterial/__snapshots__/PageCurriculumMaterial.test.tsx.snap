// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageCurriculumMaterial should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  max-width: 900px;
  width: 100%;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width: 40em) {
  .emotion-2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media screen and (min-width: 52em) {
  .emotion-2 {
    display: none;
  }
}

.emotion-3 {
  margin: 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  border: none;
  padding: 4px;
}

.emotion-4:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-4:hover,
.emotion-4:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
  fill: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow-x: auto;
}

.emotion-7 {
  display: inline-block;
  font-size: 13px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
}

.emotion-7:last-child {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-7:last-child>.splitter {
  display: none;
}

.emotion-8:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-9 {
  margin-left: 4px;
  margin-right: 4px;
}

.emotion-19 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: inline-block;
  font-size: 13px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
}

.emotion-19:last-child {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-19:last-child>.splitter {
  display: none;
}

.emotion-20 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-top: 0;
}

@media screen and (min-width: 40em) {
  .emotion-20 {
    padding-top: 0;
  }
}

@media screen and (min-width: 52em) {
  .emotion-20 {
    padding-top: 16px;
  }
}

.emotion-21 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 0;
  padding-right: 0;
}

@media screen and (min-width: 40em) {
  .emotion-21 {
    padding-left: 16px;
    padding-right: 16px;
  }
}

.emotion-22 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-surface);
  border-radius: 6px;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 0px 1px,rgba(0, 0, 0, 0.27) 0px 2px 4px -2px;
  margin-bottom: 16px;
  width: 100%;
  box-sizing: border-box;
  border-radius: 0;
}

@media screen and (min-width: 40em) {
  .emotion-22 {
    border-radius: 6px;
  }
}

.emotion-23 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 16px;
  padding-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-24 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-25 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-bottom: 4px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  line-height: 1;
  font-size: 13px;
}

.emotion-26 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
}

.emotion-27 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: var(--theme-ui-colors-surface);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  margin-left: auto;
  padding-left: 8px;
  padding-right: 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-27:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-27:hover,
.emotion-27:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-27:hover:disabled,
.emotion-27:focus:disabled {
  background-color: transparent;
  border-color: var(--theme-ui-colors-border);
}

.emotion-28 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

.emotion-29 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-surface);
  border-radius: 0;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 0px 1px,rgba(0, 0, 0, 0.27) 0px 2px 4px -2px;
  margin-left: 0;
  margin-right: 0;
  padding-bottom: 8px;
}

@media screen and (min-width: 40em) {
  .emotion-29 {
    border-radius: 6px;
    margin-left: 16px;
    margin-right: 16px;
  }
}

.emotion-30 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-31 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-bottom-width: 1px;
  border-bottom-style: solid;
  border-bottom-color: var(--theme-ui-colors-border);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-32 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin: 16px;
  font-weight: bold;
}

.emotion-33 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  border: none;
  margin-left: auto;
  margin-right: 16px;
  padding: 8px;
  color: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-33:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-33:hover,
.emotion-33:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-35 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 16px;
  margin-bottom: 8px;
}

.emotion-36 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: var(--theme-ui-colors-surface);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  margin-left: auto;
  margin-top: 16px;
  margin-right: 16px;
  color: var(--theme-ui-colors-danger);
}

.emotion-36:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-36:hover,
.emotion-36:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-36:hover:disabled,
.emotion-36:focus:disabled {
  background-color: transparent;
  border-color: var(--theme-ui-colors-border);
}

.emotion-37 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  margin-right: 8px;
  fill: var(--theme-ui-colors-danger);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <a
          class="emotion-3"
          data-cy="back-button"
          href="/dashboard/admin/curriculum/subjects?areaId=&subjectId="
        >
          <button
            class="emotion-4"
          >
            <div
              class="emotion-5"
            />
          </button>
        </a>
        <div
          class="emotion-6"
        >
          <a
            class="emotion-7"
            href="/dashboard/admin"
          >
            <span
              class="emotion-8"
            >
              Admin
            </span>
            <span
              class="splitter emotion-9"
            >
              /
            </span>
          </a>
          <a
            class="emotion-7"
            href="/dashboard/admin/curriculum"
          >
            <span
              class="emotion-8"
            >
              Curriculum
            </span>
            <span
              class="splitter emotion-9"
            >
              /
            </span>
          </a>
          <a
            class="emotion-7"
            href="/dashboard/admin/curriculum/area?areaId="
          >
            <span
              class="emotion-8"
            />
            <span
              class="splitter emotion-9"
            >
              /
            </span>
          </a>
          <a
            class="emotion-7"
            href="/dashboard/admin/curriculum/subjects?areaId=&subjectId="
          >
            <span
              class="emotion-8"
            />
            <span
              class="splitter emotion-9"
            >
              /
            </span>
          </a>
          <div
            class="emotion-19"
          >
            <span
              class="splitter"
            >
              /
            </span>
          </div>
        </div>
      </div>
      <div
        class="emotion-20"
      >
        <div
          class="emotion-21"
        >
          <div
            class="emotion-22"
          >
            <div
              class="emotion-23"
            >
              <div
                class="emotion-24"
              >
                <p
                  class="emotion-25"
                >
                  Material Name
                </p>
                <p
                  class="emotion-26"
                >
                  ...
                </p>
              </div>
              <button
                aria-label="edit-material-name"
                class="emotion-27"
              >
                <div
                  class="emotion-28"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="emotion-29"
        >
          <div
            class="emotion-30"
          >
            <div
              class="emotion-31"
            >
              <p
                class="emotion-32"
              >
                Description
              </p>
              <button
                aria-label="edit-description"
                class="emotion-33"
                data-cy="edit-description"
              >
                <div
                  class="emotion-28"
                />
              </button>
            </div>
            <div
              class="prose prose-sm max-w-none undefined emotion-35"
            >
              <p>
                Not set
              </p>
            </div>
          </div>
        </div>
        <button
          class="emotion-36"
        >
          <div
            class="emotion-37"
          />
          Delete material
        </button>
      </div>
    </div>
  </div>
</div>
`;
